sql/catalyst/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowWriter.scala

```diff
@@ -150,6 +160,13 @@ private[arrow] abstract class ArrowFieldWriter {
     valueVector.setValueCount(count)
   }
 
+  def getSizeInBytes(): Int = {
+    valueVector.setValueCount(count)
+    // Before calling getBufferSizeFor, we need to call
+    // `setValueCount`, see https://github.com/apache/arrow/pull/9187#issuecomment-763362710
+    valueVector.getBufferSizeFor(count)
+  }
```

Comment:
Hmm, based on the API doc https://arrow.apache.org/docs/java/vector.html

```
After this step, the vector enters an immutable state. In other words, we should no longer mutate it. (Unless we reuse the vector by allocating it again. This will be discussed shortly.)
```

A Java Arrow field vector after called this method, should not be modified. But I think this patch will call `getSizeInBytes` during inserting values into a vector.

Reply:
Yeah .. this has been discussed a lot, e.g., https://github.com/apache/arrow/pull/9187 but it has been many years in production without an issue so I assume this is fine ....